mappings/net/minecraft/util/collection/Weighted.mapping

```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/class_6008 net/minecraft/util/collection/Weighted
+	METHOD method_34979 getWeighting ()Lnet/minecraft/class_6007;
+	METHOD method_34980 of (Ljava/lang/Object;I)Lnet/minecraft/class_6008$class_6010;
+		ARG 1 weight
+	CLASS class_6009 Absent
+		FIELD field_29931 weighting Lnet/minecraft/class_6007;
+		METHOD <init> (I)V
+			ARG 1 weight
+		METHOD <init> (Lnet/minecraft/class_6007;)V
+			ARG 1 weighting
+	CLASS class_6010 Present
+		FIELD field_29932 object Ljava/lang/Object;
+		FIELD field_29933 weighting Lnet/minecraft/class_6007;
+		METHOD method_34981 decode (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```

Comment:
No, this is just making another codec from the current codec.

Reply:
It seems to decode the codec into a codec containing this type. I dunno... uh... `downCast`?

Reply:
```
public static <E> Codec<Weighted.Present<E>> mapCodec(Codec<E> codec) {
		return RecordCodecBuilder.create(i -> i
			.group(
				codec.fieldOf("data").forGetter(Present::getData),
				Weight.CODEC.fieldOf("weight").forGetter(Present::getWeight))
			.apply(i, Present::Present));
}
```